de.metas.adempiere.adempiere/base/src/main/java/de/metas/process/ProcessExecutionResult.java

```diff
@@ -827,58 +857,15 @@ public String toString()
 			return MoreObjects.toStringHelper(this)
 					.omitNullValues()
 					.add("records", records)
-					.add("adWindowId", adWindowIdStr)
+					.add("adWindowId", windowIdString)
 					.add("target", target)
+					.add("automaticallySetReferencingDocumentPaths", automaticallySetReferencingDocumentPaths)
 					.toString();
 		}
 
-		@Override
-		public int hashCode()
-		{
-			return Objects.hash(records, adWindowIdStr, target);
-		}
-
-		@Override
-		public boolean equals(final Object obj)
-		{
-			if (this == obj)
-			{
-				return true;
-			}
-			if (obj instanceof RecordsToOpen)
-			{
-				final RecordsToOpen other = (RecordsToOpen)obj;
-				return Objects.equals(records, other.records)
-						&& Objects.equals(adWindowIdStr, other.adWindowIdStr)
-						&& Objects.equals(target, other.target);
-			}
-			else
-			{
-				return false;
-			}
-		}
-
-		/**
-		 * @return records to open; never null or empty
-		 */
-		public List<TableRecordReference> getRecords()
-		{
-			return records;
-		}
-
 		public TableRecordReference getSingleRecord()
 		{
-			return records.get(0);
-		}
-
-		public String getWindowIdString()
-		{
-			return adWindowIdStr;
-		}
-
-		public OpenTarget getTarget()
-		{
-			return target;
+			return getRecords().get(0);
```

Comment:
given the method name, i think we should use 
`de.metas.util.collections.CollectionUtils.singleElement(Collection<T>)`
or, we should call the method getFirstRecord, if there can be >1

Reply:
as talked: there is always at least one record